File: app/src/main/java/co/meappdev/mikkipastel/ContentActivity.java
package co.meappdev.mikkipastel;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.Menu;
import android.view.MenuItem;

import co.meappdev.mikkipastel.fragment.ContentFragment;

/**
 * Created by acer on 4/30/2016.
 */
public class ContentActivity extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_content);

        if (savedInstanceState == null) {
            //1st created
            //place fragment
            getSupportFragmentManager().beginTransaction()
                    .add(R.id.contentContainer, ContentFragment.newInstance())
                    .commit();
        }
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu_share, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == R.id.action_share) {
            //onSharedContent();
            return true;
        }

        return super.onOptionsItemSelected(item);
    }

    public void onListItemClick() {
        finish();
        getSupportFragmentManager().beginTransaction()
                .remove(ContentFragment.newInstance())
                .commit();
    }

    /*private void onSharedContent() {
        // TODO Auto-generated method stub
        Intent share = new Intent(android.content.Intent.ACTION_SEND);
        share.setType("text/plain");
        String weburl = myWebView.getUrl();
        String webtitle = myWebView.getTitle();
        String shared_web = webtitle + " " + weburl;
        share.putExtra(Intent.EXTRA_TEXT, shared_web);
        startActivity(Intent.createChooser(share, "Share link"));
    }*/
}

File: app/src/main/java/com/mikkipastel/blog/activity/ContentActivity.kt
package com.mikkipastel.blog.activity

import android.content.Context
import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.mikkipastel.blog.R
import com.mikkipastel.blog.fragment.ContentFragment

class ContentActivity : AppCompatActivity() {

    companion object {

        private const val BUNDLE_BLOG_ID = "ContentActivity.BUNDLE_BLOG_ID"
        private const val BUNDLE_BLOG_TITLE = "ContentActivity.BUNDLE_BLOG_TITLE"

        fun newIntent(context: Context, blogId: String, blogTitle: String) {
            val intent = Intent(context, ContentActivity::class.java).apply {
                putExtra(BUNDLE_BLOG_ID, blogId)
                putExtra(BUNDLE_BLOG_TITLE, blogTitle)
            }
            context.startActivity(intent)
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_content)

        val blogId = intent.extras.getString(BUNDLE_BLOG_ID) ?: ""
        val blogTitle = intent.extras.getString(BUNDLE_BLOG_TITLE) ?: ""

        if (savedInstanceState == null) {
            supportFragmentManager.beginTransaction()
                    .replace(R.id.contentWebview, ContentFragment.newInstance(blogId, blogTitle))
                    .commit()
        }
    }
}